Helper/Cart.php

```diff
@@ -2174,52 +2173,42 @@ public function collectDiscounts(
-                                'discount_type'     => $this->discountHelper->convertToBoltDiscountType($couponCode), // For v1/discounts.code.apply and v2/cart.update
-                                'type'              => $this->discountHelper->convertToBoltDiscountType($couponCode), // For v1/merchant/order
-                            ];
-                            $this->logEmptyDiscountCode($couponCode, $description);
-
-                            break;
-                        case RuleInterface::COUPON_TYPE_NO_COUPON:
-                        default:
-                            $description = trim($rule->getDescription()) ? $rule->getDescription() : $rule->getName();
-                            $discounts[] = [
-                                'description'       => trim(__('Discount ') . $description),
-                                'amount'            => $roundedAmount,
-                                'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_AUTO_PROMO,
-                                'discount_type'     => $this->discountHelper->getBoltDiscountType($rule->getSimpleAction()), // For v1/discounts.code.apply and v2/cart.update
-                                'type'              => $this->discountHelper->getBoltDiscountType($rule->getSimpleAction()), // For v1/merchant/order
-                            ];
-
-                            break;
-                    }
-                    $totalAmount -= $roundedAmount;
+                $roundedAmount = CurrencyUtils::toMinor($ruleDiscountAmount, $currencyCode);
+                switch ($rule->getCouponType()) {
+                    case RuleInterface::COUPON_TYPE_SPECIFIC_COUPON:
+                    case RuleInterface::COUPON_TYPE_AUTO:
+                        $couponCode = $quote->getCouponCode();
+                        $ruleDescription = $rule->getDescription();
+                        $description = $ruleDescription !== '' ? $ruleDescription : 'Discount (' . $couponCode . ')';
+                        $discountType = $this->discountHelper->convertToBoltDiscountType($couponCode);
+                        $discounts[] = [
+                            'description'       => $description,
+                            'amount'            => $roundedAmount,
+                            'reference'         => $couponCode,
+                            'discount_category' => Discount::BOLT_DISCOUNT_CATEGORY_COUPON,
+                            'discount_type'     => $discountType, // For v1/discounts.code.apply and v2/cart.update
+                            'type'              => $discountType, // For v1/merchant/order
+                        ];
+                        $this->logEmptyDiscountCode($couponCode, $description);
+
+                        break;
+                    case RuleInterface::COUPON_TYPE_NO_COUPON:
+                    default:
+                        $discountType = $this->discountHelper->convertToBoltDiscountType('');
```

Comment:
@dumega Why do you empty the parameter in method  **convertToBoltDiscountType**? 
Is $this->discountHelper->convertToBoltDiscountType($rule->getSimpleAction())  wrong?

Reply:
After consideration, I think $this->discountHelper->getBoltDiscountType is more suitable for getting bolt discount type by action of rule, and I have updated the code. please review again